### PR TITLE
AVI IPv4/IPv6 Dual Stack Support

### PR DESCRIPTION
**Pull request by chenlin07**
**What this PR does / why we need it**:

Support AVI IPv4/IPv6 Dual Stack 

**Which issue(s) this PR fixes**:

Fixes #

**Describe testing done for PR**:

**Special notes for your reviewer**:

**Release note**:

```release-note

```
**New PR Checklist**

- [ ] Ensure PR contains only public links or terms
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Squash the commits in this branch before merge to preserve our git history
- [ ] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [ ] Add appropriate [labels](https://github.com/vmware-tanzu/load-balancer-operator-for-kubernetes/labels) according to what type of issue is being addressed.

### CODE REVIEW

File: pkg/haprovider/haprovider.go [hunk 1]
```diff
@@ -270,11 +270,13 @@ func (r *HAProvider) syncEndpointMachineIP(address corev1.EndpointAddress, machi
 			if machineAddress.Type != clusterv1.MachineExternalIP {
 				continue
 			}
-			if net.ParseIP(machineAddress.Address) != nil && net.ParseIP(machineAddress.Address).To4() != nil {
-				address.IP = machineAddress.Address
-				r.log.Info("sync endpoints object, update machine: " + machine.Name + "'s ip to:" + address.IP)
-			} else {
-				r.log.Info(machineAddress.Address + " is not a valid IPv4 address")
+			if net.ParseIP(machineAddress.Address) != nil {
```
Comment (XudongLiuHarold) [inline pkg/haprovider/haprovider.go:273]: This condition can be changed to 
```go
if net.ParseIP(machineAddress.Address) != nil {
}
```

File: pkg/haprovider/haprovider.go [hunk 2]
```diff
@@ -327,16 +329,18 @@ func (r *HAProvider) addMachineIpToEndpoints(endpoints *corev1.Endpoints, machin
 			continue
 		}
 		// check machineAddress.Address is valid
-		// only support IPv4 for now
-		if net.ParseIP(machineAddress.Address) != nil && net.ParseIP(machineAddress.Address).To4() != nil {
-			newAddress := corev1.EndpointAddress{
-				IP:       machineAddress.Address,
-				NodeName: &machine.Name,
+		// Support IPv4 and IPv6
+		if net.ParseIP(machineAddress.Address) != nil {
```
Comment (XudongLiuHarold) [inline pkg/haprovider/haprovider.go:333]: similarly for this one

**Review by XudongLiuHarold — COMMENTED** [review]
Generally looks good to me, but we need to test it e2ely

File: pkg/haprovider/haprovider.go (hunk 1 above)
Comment (chenlin07) [reply pkg/haprovider/haprovider.go:273]: I thought these two lines are same?

File: pkg/haprovider/haprovider.go [hunk 3]
```diff
@@ -270,11 +270,9 @@ func (r *HAProvider) syncEndpointMachineIP(address corev1.EndpointAddress, machi
 			if machineAddress.Type != clusterv1.MachineExternalIP {
 				continue
 			}
-			if net.ParseIP(machineAddress.Address) != nil && net.ParseIP(machineAddress.Address).To4() != nil {
+			if net.ParseIP(machineAddress.Address) != nil {
 				address.IP = machineAddress.Address
 				r.log.Info("sync endpoints object, update machine: " + machine.Name + "'s ip to:" + address.IP)
-			} else {
```
Comment (XudongLiuHarold) [inline pkg/haprovider/haprovider.go:276]: Why remove the else, it also could be an invalid IP address, not v4 or v6

File: pkg/haprovider/haprovider.go [hunk 4]
```diff
@@ -327,16 +325,14 @@ func (r *HAProvider) addMachineIpToEndpoints(endpoints *corev1.Endpoints, machin
...
-		if net.ParseIP(machineAddress.Address) != nil && net.ParseIP(machineAddress.Address).To4() != nil {
+		// Support IPv4 and IPv6
+		if net.ParseIP(machineAddress.Address) != nil {
 			newAddress := corev1.EndpointAddress{
 				IP:       machineAddress.Address,
 				NodeName: &machine.Name,
 			}
 			endpoints.Subsets[0].Addresses = append(endpoints.Subsets[0].Addresses, newAddress)
 			break
-		} else {
```
Comment (XudongLiuHarold) [inline pkg/haprovider/haprovider.go:338]: same here ^

**Review by XudongLiuHarold — COMMENTED** [review]
generally LGTM, minor changes. Please make sure the pre-merge pipeline can pass, thx!

File: api/v1alpha1/akodeploymentconfig_types.go [hunk 5]
```diff
@@ -365,7 +365,8 @@ type ControlPlaneNetwork struct {
 // VIPNetwork describes a VIPNetwork in the adc file
 type VIPNetwork struct {
 	NetworkName string `json:"networkName"`
-	CIDR        string `json:"cidr"`
+	CIDR        string `json:"cidr,omitempty"`
+	V6CIDR      string `json:"v6cidr,omitempty"`
```
[the remark this answers is not in the document]
Comment (chenlin07) [reply api/v1alpha1/akodeploymentconfig_types.go:369]: No, the crd is still same

**Review by XudongLiuHarold — APPROVED** [review]
LGTM

but we still need to verify every support scenarios before we merge this

**Pull request closed without merging**